plugins/core/editor/routes/locales.js

```diff
@@ -51,25 +60,35 @@ module.exports = {
       },
       handler: (request, h) => {
         // compute a translation.json file for use by i18next for the given language
-        let translations = {};
+        const translations = {}
+        let previewSizes
+
+        const criteria = request.auth ? request.auth.credentials : {};
+
+        // check for the get function to be backwards compatible if the config object gets
+        // passed directly instead of providing a Confidence getter
+        if (options.editorConfig.get instanceof Function) {
+          previewSizes = options.editorConfig.get('/previewSizes', criteria)
+        } else {
+          previewSizes = options.editorConfig.previewSizes
+        }
 
-        const previewSizes = options.editorConfig.previewSizes;
         if (previewSizes) {
-          translations.preview = {};
-          for (let previewSizeName in previewSizes) {
-            const previewSize = previewSizes[previewSizeName];
+          translations.preview = {}
+          for (const previewSizeName in previewSizes) {
+            const previewSize = previewSizes[previewSizeName]
             if (
-              previewSize.hasOwnProperty("label_locales") &&
+              previewSize.hasOwnProperty('label_locales') &&
               previewSize.label_locales.hasOwnProperty(request.params.lng)
             ) {
               translations.preview[previewSizeName] =
-                previewSize.label_locales[request.params.lng];
+                previewSize.label_locales[request.params.lng]
             }
           }
         }
 
-        return translations;
+        return translations
       }
-    };
+    }
   }
-};
+}
```

Comment:
A separate commit with the linting-specific changes would be good. That makes it easier to review the code.

Reply:
done, I kept the style from before to not diverge the code style from the rest of the Q server code base.